run: Get email from a user in the specified format.

preetmishra opened the pull request:
This commit specifies the Email format according to the values set in
`require_email_format_usernames` and `email_auth_enabled`.
It also performs a validation check to see if the Email entered by a user
is valid.

Fixes #491.

amanagr commented:
Thanks @preetmishra. This looks great. Next step would be to add test cases for the changes you have made in `tests/cli/test_run.py`.

preetmishra commented:
@amanagr Thank you. I have added a commit that adds test for the changes. It is ready for review.

neiljp commented:
@preetmishra I've been quite busy with other things recently, but am taking a look now.

preetmishra commented on zulipterminal/cli/run.py:
```diff
@@ -66,10 +67,33 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
+    if not require_email_format_usernames and email_auth_enabled:
+        email = input(in_color('blue', 'Email or Username: '))
+    elif not require_email_format_usernames:
+        email = input(in_color('blue', 'Username: '))
```
Yes, that'd be a lot cleaner. I'll change that.

preetmishra commented on zulipterminal/cli/run.py:
```diff
@@ -66,10 +67,33 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
+        while True:
+            email = input(in_color('blue', 'Email: '))
+            # RegEx: Checks whether Email address is of format '*@*.*'
+            regex_pattern = r'[^@]+@[^@]+\.[^@]+'
```
No, I wrote it. But you're right It'll end up failing for those two scenarios. I'll add a `TODO` there for now.

preetmishra commented:
I agree. I haven't written much test code before. I am willing to put in the effort to write better tests.

preetmishra commented:
@neiljp I have resolved the conflict and changed `get_email` to `get_login_id` as the value returned by the function can either be an email address or a username so `get_login_id` makes more sense. I have also changed variables related to it. Further, I have made the condition block cleaner using a label variable and added TODO for the email validation. 
I have removed the other test commit (tests: run:) from this PR for now as suggested.
It is ready for review.

neiljp commented:
@preetmishra Thanks for working on this :+1: I've merged this commit with a change to the commit message (it doesn't fix the client side email check, so I removed the fixes part too), and followed it with refactor and test commits on master which should give an idea of how to move forward with more error handling.
